server/controllers/finance/reports/balance/index.js

```diff
@@ -105,37 +102,39 @@ function processAccounts(balances, accounts, totals) {
     accounts[row.number] = account;
   });
 
-  Object.keys(accounts).forEach(function (item) {
+  Object.keys(accounts).forEach((item) => {
     accounts[item].endDebit = 0;
     accounts[item].endCredit = 0;
-    let sold = (accounts[item].beginDebit || 0 - accounts[item].beginCredit || 0) + (accounts[item].middleDebit - accounts[item].middleCredit);
+    const sold = (accounts[item].beginDebit || 0 - accounts[item].beginCredit || 0)
+      + (accounts[item].middleDebit - accounts[item].middleCredit);
     if (sold < 0) {
       accounts[item].endCredit = sold * -1;
     } else {
-     accounts[item].endDebit = sold;
+      accounts[item].endDebit = sold;
     }
   });
 
   // process for getting totals
-  totals = Object.keys(accounts).reduce(function (totals, key) {
-    let account = accounts[key];
-    totals.beginDebit += (account.beginDebit || 0);
-    totals.beginCredit += (account.beginCredit || 0);
-    totals.middleDebit += (account.middleDebit || 0);
-    totals.middleCredit += (account.middleCredit || 0);
-    totals.endDebit += (account.endDebit || 0);
-    totals.endCredit += (account.endCredit || 0);
-    return totals;
-  }, {
-    beginDebit   : 0,
-    beginCredit  : 0,
-    middleDebit  : 0,
-    middleCredit : 0,
-    endDebit     : 0,
-    endCredit    : 0,
-  });
+  const vtotals = Object.keys(accounts)
```

Comment:
I thought that this report/ code had been replaced with MySQL maths logic? This code should be removed as soon as possible.